### DISCUSSION TITLE
`tar_render()` from a static branch

### DISCUSSION THREAD

**Discussion by 11rchitwood** [ropensci/tarchetypes#104] [Help] [answered]
Is there any good way to render from a static branch? I'd like to read in target into an Rmd file. Here's a small example pipeline.

``` r
library(targets)
library(tarchetypes)

targets <- tar_map(
  values = data.frame(sp = levels(iris$Species)),
  tar_target(data, dplyr::filter(iris, Species == sp)),
  tar_render(report, "report.Rmd")
)
```

And the corresponding Rmd
```
lines <- c(
  "---",
  "title: report.Rmd source file",
  "output_format: html_document",
  "---",
  "Assume these lines are in report.Rmd.",
  "```{r}",
  "targets::tar_read(data)",
  "```"
)
```

I'd like to be able to have the Rmd as is, but the actual target be data_setosa, etc.

**Comment by wlandau** [comment, marked as answer]
I think `tar_render_rep()` and its dynamic branching over R Markdown parameters would be a good fit for this use case. It's trickier with static branching.

**Reply by 11rchitwood** [reply]
I thought that might be it. Thanks!
